src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -356,14 +371,57 @@ const Hole::ThreadDescription Hole::threadDescription[][171] =
 
 };
 
+const double Hole::metricHoleDiameters[35][4] =
+{
+    /* ISO metric according to ISO 273 */
+    // {screw diameter, close, standard, coarse}
+        { 1.0,      1.1,    1.2,    1.3},
+        { 1.2,      1.3,    1.4,    1.5},
+        { 1.4,      1.5,    1.6,    1.8},
+        { 1.6,      1.7,    1.8,    2.0},
+        { 1.8,      2.0,    2.1,    2.2},
+        { 2.0,      2.2,    2.4,    2.6},
+        { 2.5,      2.7,    2.9,    3.1},
+        { 3.0,      3.2,    3.4,    3.6},
+        { 3.5,      3.7,    3.9,    4.2},
+        { 4.0,      4.3,    4.5,    4.8},
+        { 4.5,      4.8,    5.0,    5.3},
+        { 5.0,      5.3,    5.5,    5.8},
+        { 6.0,      6.4,    6.6,    7.0},
+        { 7.0,      7.4,    7.6,    8.0},
+        { 8.0,      8.4,    9.0,    10.0},
+        { 10.0,     10.5,   11.0,   12.0},
+        { 12.0,     13.0,   13.5,   14.5},
+        { 14.0,     15.0,   15.5,   16.5},
+        { 16.0,     17.0,  	17.5,   18.5},
+        { 18.0,     19.0,  	20.0,   21.0},
+        { 20.0,     21.0,  	22.0,   24.0},
+        { 22.0,     23.0,  	24.0,   26.0},
+        { 24.0,     25.0,  	26.0,   28.0},
+        { 27.0,     28.0,  	30.0,   32.0},
+        { 30.0,     31.0,  	33.0,   35.0},
+        { 36.0,     37.0,  	39.0,   42.0},
+        { 39.0,     40.0,  	42.0,   45.0},
+        { 42.0,     43.0,  	45.0,   48.0},
+        { 45.0,     46.0,  	48.0,   52.0},
+        { 48.0,     50.0,  	52.0,   56.0},
+        { 52.0,     54.0,  	56.0,   62.0},
+        { 56.0,     58.0,  	62.0,   66.0},
+        { 60.0,     62.0,  	66.0,   70.0},
+        { 64.0,     66.0,  	70.0,   74.0},
+        { 68.0,     70.0,  	77.0,   78.0}
+};
+
 /* ISO coarse metric enums */
 std::vector<std::string> Hole::HoleCutType_ISOmetric_Enums  = { "None", "Counterbore", "Countersink", "Cheesehead (deprecated)", "Countersink socket screw (deprecated)", "Cap screw (deprecated)" };
-const char* Hole::ThreadSize_ISOmetric_Enums[]   = { "M1.6",  "M2",  "M2.5",  "M3",
-                                                     "M3.5",  "M4",  "M5",    "M6",
-                                                     "M8",    "M10", "M12",   "M14",
-                                                     "M16",   "M20", "M22",   "M24",
-                                                     "M27",   "M30", "M36",   "M42",
-                                                     "M48",   "M56", "M64",   "M68", NULL };
+const char* Hole::ThreadSize_ISOmetric_Enums[]   = { "M1.0", "M1.1", "M1.2", "M1.4", "M1.6",

```

Comment:
Is an M1 really called an "M1.0"?

Reply:
Thanks! You are really good in finding these small issues. I fixed this now already.